packages/ui/src/lib/link/Link.svelte

```diff
@@ -1,18 +1,21 @@
 <script lang="ts">
-import { createEventDispatcher, onMount } from 'svelte';
+import type { IconDefinition } from '@fortawesome/free-regular-svg-icons';
+import { onMount, type Snippet } from 'svelte';
 import Fa from 'svelte-fa';
 
 import { isFontAwesomeIcon } from '../utils/icon-utils';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export let icon: any = undefined;
+interface Props {
+  icon?: IconDefinition;
+  class?: string;
+  children: Snippet;
+  onclick: () => void;
+  'aria-label'?: string;
+}
 
-let iconType: string | undefined = undefined;
+let { icon = undefined, class: classes = '', 'aria-label': ariaLabel = '', children, onclick }: Props = $props();
 
-const dispatch = createEventDispatcher<{ click: undefined }>();
-export let onclick: () => void = () => {
-  dispatch('click');
-};
```

Comment:
issue: removing the dispatcher will break all components

as intermediate step we introduced both the callback and the dispatcher (if callback is provided, the dispatch event is not thrown)

removing that part will break all existing apps using Link

I think we should keep this approach for a while, the time to migrate in apps the move to the callback rather than the event